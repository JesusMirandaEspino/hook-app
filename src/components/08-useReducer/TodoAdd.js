import React from 'react'
import { UseForm } from '../../Hooks/UseForm';

export const TodoAdd = ( { handleAddTodo } ) => {

    const [  { description } , handleInpuntChange, reset ]  = UseForm( {
        description: ''
    } );


    const handleSubmit = (e) => {
        e.preventDefault();

        if( description.trim() <= 1 ){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        };

        handleAddTodo( newTodo );
        reset();

    }

    return (
        <>
            <form  onSubmit={ handleSubmit } >

                <label for="description" > Ingresa un elemento </label> 
                <input type="text"  name="description" onChange={ handleInpuntChange }  value={ description }  />

                <button type="submit" className="btn btn-primary mt-1" >Agregar</button>

            </form>            
        </>
    )
}
