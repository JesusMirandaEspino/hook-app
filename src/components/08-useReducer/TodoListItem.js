import React from 'react'

export const TodoListItem = ( { todo, index, HandleDelete, handleToggle }  ) => {
    return (
            <li key={ todo.id }  className="list-group-item" >
                <p className={  ` ${ todo.done && 'complete' } ` } onClick={ () => handleToggle(todo.id) }  >  { index + 1 } .- {  todo.desc } </p>
                
                <button className="btn btn-danger" onClick={ () => HandleDelete(todo.id) }    >Borrar</button>

            </li>
    )
}
