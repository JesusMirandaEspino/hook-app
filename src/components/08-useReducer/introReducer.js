const initialState = [{
    id: 1,
    todo: 'Curso Angular',
    done: false
}];

const todoReducer = ( state = initialState, action ) => {

    if( action.type == 'agregarTodoAction' ){}

    return state;
};


let todos = todoReducer();

const newTodo = {
    id: 2,
    todo: 'Curso React',
    done: false
};

const agregarTodoAction = {
    type: 'agregar',
    payload: newTodo
};


todos = todoReducer(todos, agregarTodoAction);