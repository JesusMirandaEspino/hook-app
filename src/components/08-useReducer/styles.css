body{
    padding:50px;
}


.complete{
    text-decoration: line-through;
}

p{
    cursor: pointer;
}