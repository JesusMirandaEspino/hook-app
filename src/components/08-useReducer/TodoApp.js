import React, { useReducer } from 'react';
import { TodoReducer } from './TodoReducer'
import './styles.css';

export const TodoApp = () => {

    const initialState = [{

        id: new Date().getTime(),
        desc: 'Aprender React',
        done: false
    }];


    const [  todo ] = useReducer( TodoReducer, initialState  );


    

    return (
        <div>
            <h1>Todo App</h1>
            <hr/>

            <ul>

                <li>Hola</li>
                <li>Mundo</li>
                <li>Como</li>
                <li>Estan</li>

            </ul>

        </div>
    )
}
