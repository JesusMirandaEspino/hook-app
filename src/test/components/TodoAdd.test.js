import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { TodoAdd } from '../../components/08-useReducer/TodoAdd';


describe( 'Pruebas con <TodoAdd />', () => {

    const  handleAddTodo = jest.fn();

    const wrappers = shallow( <TodoAdd  handleAddTodo={ handleAddTodo}  /> );

    test( 'Debe de mostrar correctamente', () => {

        expect( wrappers ).toMatchSnapshot();

    });

    test( 'No debe de llamar addTodo', () => {

        const formSumit = wrappers.find( 'form' ).prop('onSubmit');

        formSumit( { preventDefault(){} } );

        expect( handleAddTodo ).toHaveBeenCalledTimes(0);

    });


});