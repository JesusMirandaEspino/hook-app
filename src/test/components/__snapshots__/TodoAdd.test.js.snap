// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <TodoAdd /> Debe de mostrar correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <label
      for="description"
    >
       Ingresa un elemento 
    </label>
    <input
      name="description"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn btn-primary mt-1"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
